### ISSUE TITLE
Error calculating simple IF formula which MS-Excel can do

### ISSUE THREAD

**Issue by AnkurRaval** [ClosedXML/ClosedXML#1530]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.95.3

**What is the current behavior?**

In the attached sheet, there's a formula in column B. The formula is **=IF(A2="Lenovo","100",IF(D2=0,"900","700"))**
Column D has Credit Amount value. In Microsoft excel, if D2 cell value is blank then also it considers it as D2=0 and gives me result as 900 however ClosedXML for the same formula gives me calculated result as 700.
I've attached a code below which converts excel data to datatable. The AccountNumber column there should return 4 rows of values (900,900,100,100) but with closedxml it returns 700,700,100,100.

**What is the expected behavior or new feature?**

Formula mentioned in Current behavior should act similar to excel and give me  (900,900,100,100) in AccountNumber column.
**Is this a regression from the previous version?**

No

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
Using System.Data;


public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a 
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code 
    //   samples on the wiki are good examples of the terseness that I want. Don't
    //   post your full application.
var dt = ExcelToDatatable(filepath,0);
}
public static DataTable ExcelToDatatable(string filepath, int NumofRowsToSkip)
{
    DataTable dtExcel = new DataTable();
    using (new HUB.Impersonation())
    using (XLWorkbook workBook = new XLWorkbook(filepath))
    {
        IXLWorksheet workSheet = workBook.Worksheet(1);
        bool firstRow = true;
        int amountIndex = 0;
        int index = 0;
        foreach (IXLRow row in workSheet.Rows())
        {
            if (row.RowNumber <= NumofRowsToSkip)
                continue;
            if (firstRow)
            {
                foreach (IXLCell cell in row.Cells())
                    dtExcel.Columns.Add(cell.Value.ToString());
                firstRow = false;
            }
            else
            {
                // Add rows to DataTable.
                dtExcel.Rows.Add();
                for (var i = 0; i <= dtExcel.Columns.Count - 1; i++)
                {
                    IXLCell cell = row.Cell(i + 1);
                    dtExcel.Rows(dtExcel.Rows.Count - 1)(i) = cell.Value;
                }
            }
        }
    }
    return dtExcel;
}



```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[ClosedXMLIssueforGIT.xlsx](https://github.com/ClosedXML/ClosedXML/files/5261648/ClosedXMLIssueforGIT.xlsx)
[ClosedXMLIssueforGIT.xlsx](https://github.com/ClosedXML/ClosedXML/files/5261632/ClosedXMLIssueforGIT.xlsx)

**Comment by igitur** [comment]
ClosedXML currently treats empty cells as having a value of `"" `, an empty string. You're comparing D2, which is empty, to 0 and that test currently fails. There are plans to fix this, but it definitely isn't an easy fix. Try changing the `D2=0` to `D2="" `

**Comment by AnkurRaval** [comment]
@igitur What you said makes sense. I've no idea how closedxml works as I recently started using it. But in simple terms, if it is able to identify that the column is a numeric column then it can simply treat blanks as 0.
But Yes, I'll advise my users to change the formula for now.

**Comment by igitur** [comment]
Empty cells are not always treated as 0. If they were, then `D2=""` would be false.

**Comment by AnkurRaval** [comment]
Hi @igitur 
Could you please tell me what's the best way to read values from formula cells?
Currently I've created an extension method which first tries to return cell.value and if it fails in some cases then from Catch block I return cell.ValueCached. Now for this particular excel that I've attached, I'm getting incorrect values (700) compared to excel in the **Value** and **Cachedvalue** fields; however, **ValueCached** has correct value(900). So I'm not sure when to use which one of these 3 methods.

**Comment by Pankraty** [comment]
CachedValue is used for storing results of formula computation. If you saved the file in Excel then CachedValue contains the correct value, but this is unreliable because there are no guarantees that this value is saved - it may be empty too. 
If you call Value in ClosedXML then the CachedValue is overwritten with the (incorrectly) calculated value and becomes quite useless. 
So, your solution works fine if there is a formula that is unsupported by ClisedXML but it cannot help if the computation works incorrectly without exceptions.

**Comment by igitur** [comment]
@AnkurRaval Please test the debug build at https://ci.appveyor.com/project/ClosedXML/closedxml/builds/37851782/artifacts and let me know if this solves your problem.

**Comment by jahav** [comment]
The issue of corection and conversion has bee solved by #1812 , so the IF formula the original comment works. However, the fundamental correctness requires #1894 and to rewrite functions so they work as they should.

Either way, the formula `IF(A2="Lenovo","100",IF(D2=0,"900","700"))` works and the issue is inactive for 2 years. Closing as duplicate of #1812.
